lib/tasks/queue.rake

```diff
@@ -29,31 +30,35 @@ namespace :queue do
   end
 
   desc "Add published editions to the message queue by document type"
-  task :requeue_document_type, [:document_type] => :environment do |_, args|
-    document_type = args[:document_type]
-    raise ValueError("expecting document_type") unless document_type.present?
+  task :requeue_document_type, [] => :environment do |_, args|
+    document_types = args.extras
 
-    # Restrict scope to stuff that's live (published or unpublished)
-    # Unpublished content without a content store representation won't
-    # be returned, but we're not interested in this content.
-    scope = Edition
-      .with_document
-      .with_unpublishing
-      .where(document_type: document_type)
-      .where(content_store: :live)
-      .select(:id)
+    raise(ValueError, "expecting document_type") unless document_types.present?
 
-    version = Event.maximum(:id)
+    document_types.each do |document_type|
+
+      # Restrict scope to stuff that's live (published or unpublished)
+      # Unpublished content without a content store representation won't
+      # be returned, but we're not interested in this content.
+      scope = Edition
+        .with_document
+        .with_unpublishing
+        .where(document_type: document_type)
+        .where(content_store: :live)
+        .select(:id)
+
+      version = Event.maximum(:id)
```

Comment:
you can take this outside of the loop. The purpose of the version is just to make sure that any later publishing events will always have a later version, so that if the represent downstream takes long a time to process, it can't overwrite updates that happened since it was triggered.

Reply:
@MatMoore have moved